powerline/segments/i3wm.py

```diff
@@ -52,15 +52,18 @@ def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
 
 	return [
 		{
-			'contents': w['name'][strip:],
+			'contents': w.name[strip:],
 			'highlight_groups': workspace_groups(w)
 		}
 		for w in get_i3_connection().get_workspaces()
-		if ((not only_show or any(w[typ] for typ in only_show))
-		    and (not output or w['output'] == output))
+		if ((not only_show or any(getattr(w, typ) for typ in only_show))
+		    and (not output or w.output == output))
 	]
 
 
+class Dummy:
```

Comment:
Why do we need this `Dummy` class?

Reply:
ah, I knew I should've started from scratch ;)